src/transformer.js

```diff
@@ -115,6 +141,28 @@ function _transform( xslStr, xmlDoc, xsltParams ) {
     } );
 }
 
+/**
+ * @param {string} src
+ * @return {string}
+ */
+function _escapeSrcAttribute( src ) {
```

Comment:
Where do these steps come from? I think you mentioned the XML lib takes care of escaping in cases where the jr:// URIs are src attributes — does this match? Is there any way to use that implementation?

Reply:
- Checking for fully qualified URLs handles escaping `jr:` URLs to return them fully qualified.
- Falling back to `file:///` handles media mappings which are relative paths.
- Wrapping in a `URL` natively performs the relevant escaping while preserving valid URL characters.
- Checking the leading `/` is probably unnecessary, but respects the presence or absence of leading slashes in the unescaped media mapping.

The XML lib does escape `src` attributes, but only in an HTML document context where the DOM equivalent would do the same. That functionality doesn’t exist on the XML model document, also consistent with the DOM. This ensures `jr:` URLs and their media mappings are always escaped consistently in every context.

Importantly, it ensures downstream usage will be able to use consistent escaping without depending on the XML lib. It’s unlikely they behave differently, I’d assume they’re making the same V8 calls, but it seemed both prudent and pragmatic to use the same API everywhere.

Reply:
> Falling back to file:/// handles media mappings which are relative paths.

Hmmm. Ok, so if the input is `/foo/bar/baz`, adding `file:///` WITH THREE `/` means that the `href` which is returned will be `/foo/bar/baz`? And then the things about stripping the leading `/` are because `foo/bar/baz` would also then be `/foo/bar/baz`? Unit tests would really help illustrate these cases.

If I change the current default in the test to `/images/hallo spaceboy/under pressure.png` or `images/hallo spaceboy/under pressure.png` it looks like both are passed through verbatim without any escape so maybe something isn't going quite right?

Reply:
Right, `/^[a-zA-Z]+:\/\//.test( value )` fails so the escaped value is unused. Hrm. I still think unit tests for `_processBinaryDefaults` would be helpful to illustrate the cases but this shows they're not really enough.

Reply:
> Hmmm. Ok, so if the input is /foo/bar/baz, adding file:/// WITH THREE / means that the href which is returned will be /foo/bar/baz? And then the things about stripping the leading / are because foo/bar/baz would also then be /foo/bar/baz?

This is correct.

> Unit tests would really help illustrate these cases.

The existing tests using leading slashes are actually how I caught this. But I'm happy to add another test *specifically* demonstrating this difference so it's more explicit.

> If I change the current default in the test to /images/hallo spaceboy/under pressure.png or images/hallo spaceboy/under pressure.png it looks like both are passed through verbatim without any escape so maybe something isn't going quite right?

Yep, `_processBinaryDefaults` was not using the media mapping. I missed these comments earlier, but caught this while we were discussing the other comment below. It's fixed in 2a0b72c.

> Right, /^[a-zA-Z]+:\/\//.test( value ) fails so the escaped value is unused. Hrm. I still think unit tests for _processBinaryDefaults would be helpful to illustrate the cases but this shows they're not really enough.

This is also now exercised in my new test, but again I'm happy to add a more explicit test.